src/pages/create-or-edit-lesson/CreateOrEditLesson.tsx

```diff
@@ -165,44 +155,71 @@ const CreateOrEditLesson = () => {
   } = useForm<LessonData>({
     initialValues,
     validations,
-    onSubmit: id ? fetchEditedLesson : fetchAddLesson,
+    onSubmit: () => {
+      if (id) {
+        fetchEditedLesson()
+      } else {
+        fetchAddLesson()
+      }
+    },
     submitWithData: true
   })
 
-  const getLesson = (id?: string): Promise<AxiosResponse> => {
-    return ResourceService.getLesson(id)
-  }
+  const getLesson = useCallback(() => {
+    if (id) {
+      return ResourceService.getLesson(id)
+    }
 
-  const handleResponseLesson = (lesson: LessonData) => {
-    for (const key in data) {
-      const validKey = key as keyof LessonData
-      handleNonInputValueChange(validKey, lesson[validKey])
+    return defaultResponse
+  }, [id])
+
+  const {
+    isLoading,
+    error,
+    data: lesson
+  } = useQuery({
+    queryKey: ['lesson', id],
+    queryFn: getLesson,
+    options: {
+      initialData: defaultResponse,
+      enabled: Boolean(id)
     }
-  }
+  })
+
+  const editLesson = useCallback(async () => {
+    if (id) {
+      return ResourceService.editLesson(data, id)
+    }
+
+    return Promise.resolve(defaultResponse)
+  }, [data, id])
```

Comment:
If the backend doesn't return something, you can just conditionaly call service without any return value
```suggestion
  const editLesson = useCallback(async () => {
    if (id) {
      await ResourceService.editLesson(data, id)
    }
  }, [data, id])
```